test/ integration.test.js

```diff
@@ -0,0 +1,79 @@
+const path = require('path');
+const Generator = require('@asyncapi/generator');
+const { readFile } = require('fs').promises;
+const crypto = require('crypto');
+const fetch = require('node-fetch');
+
+const MAIN_TEST_RESULT_PATH = path.join('test', 'temp', 'integrationTestResult');
+
+const generateFolderName = () => {
+    // you always want to generate to new directory to make sure test runs in clear environment
+    return path.resolve(MAIN_TEST_RESULT_PATH, crypto.randomBytes(4).toString('hex'));
```

Comment:
What about using `Math.random` here instead of `crypto` module? I think it will be simpler and faster too.